src/arch/USB/USBDriver_Impl.cpp

```diff
@@ -0,0 +1,34 @@
+#include "global.h"
+#include "USBDriver_Impl.h"
+#include "arch/arch_default.h"
+
+#include "arch/USB/USBDriver_Impl_Libusb.h"
+
+USBDriver_Impl::USBDriver_Impl()
+{
+	/* no-op */
+}
+
+USBDriver_Impl::~USBDriver_Impl()
+{
+	/* no-op */
+}
+
+/* libusb has working implementations on the three major systems - Linux,
+ * Mac, and Windows - but they may not perform as well as the native APIs.
+ * If we have a different API, prefer that first. */
+
+USBDriver_Impl* USBDriver_Impl::Create()
+{
+#if defined(HAS_USBDRIVER_IMPL_LIBUSB)
+	return new USBDriver_Impl_Libusb;
+#endif
```

Comment:
I think we could just always use libusb for all platforms. That way we could get rid of the dispatch mechanism and remove two of the files.